share/functions/prompt_pwd.fish

```diff
@@ -8,18 +8,34 @@ function prompt_pwd --description "Print the current working directory, shortene
         return 0
     end
 
-    # This allows overriding fish_prompt_pwd_dir_length from the outside (global or universal) without leaking it
+    # This allows overriding fish_prompt_pwd_dir_length and fish_prompt_pwd_depth from the outside (global or universal) without leaking it
     set -q fish_prompt_pwd_dir_length
     or set -l fish_prompt_pwd_dir_length 1
+  
+    set -q fish_prompt_pwd_depth
+    or set -l fish_prompt_pwd_depth 0
 
     # Replace $HOME with "~"
     set realhome ~
     set -l tmp (string replace -r '^'"$realhome"'($|/)' '~$1' $PWD)
+    # Split $temp to an array(for the trim depth only) 
+    set -l tempArr (string split '/' $tmp)
 
     if [ $fish_prompt_pwd_dir_length -eq 0 ]
-        echo $tmp
+
+        if [ $fish_prompt_pwd_depth -gt (count $tempArr) -o $fish_prompt_pwd_depth -eq 0 ] #dont trim, dont shorten
+            echo $tmp 
+        else #trim, dont shorten
+            echo (string join '/' $tempArr[(math "-1 * "$fish_prompt_pwd_depth"")..-1]) 
+        end
     else
-        # Shorten to at most $fish_prompt_pwd_dir_length characters per directory
-        string replace -ar '(\.?[^/]{'"$fish_prompt_pwd_dir_length"'})[^/]*/' '$1/' $tmp
+
+        if [ $fish_prompt_pwd_depth -gt (count $tempArr) -o $fish_prompt_pwd_depth -eq 0 ] #dont trim, shorten
+            # Shorten to at most $fish_prompt_pwd_dir_length characters per directory
+            string replace -ar '(\.?[^/]{'"$fish_prompt_pwd_dir_length"'})[^/]*/' '$1/' $tmp
+        else #trim, shorten
+            set -l trimmed_pwd (string join '/' $tempArr[(math "-1 * "$fish_prompt_pwd_depth"")..-1]) 
```

Comment:
`math` actually isn't needed here at all - "-$fish_prompt_pwd_depth" would work.

Alternatively, you could erase the earlier elements from the array but I believe that would make the logic a bit more complicated.

Reply:
Or does this allow negative values for $fish_prompt_pwd_depth?

Reply:
> Or does this allow negative values for $fish_prompt_pwd_depth?

Nah I don't think anyone will use a reverse depth shortening, this is unintended. I will try to fix this